### PR TITLE
Note S3 versioning requirement

### PR DESCRIPTION
**Pull request by alexfornuto**
Addresses the documentation component of #10698 by documenting the need for versioned S3 buckets for session recodings.

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]
Looks good. Do you think it makes sense to add a little more color? I was thinking something like:

"Teleport does this by ensuring it always looks at the oldest version of the recording."

**Comment by zmb3** [comment]
Thanks for this! You can probably backport this to 8/9 as well as this is not new behavior.

File: docs/pages/reference/backends.mdx [hunk 1]
```diff
@@ -195,8 +195,14 @@ teleport:
...
 S3 buckets can only be used as storage for the recorded sessions. S3 cannot
-store the audit log or the cluster state. Below is an example of how to
-configure a Teleport auth server to store the recorded sessions in an S3 bucket.
+store the audit log or the cluster state.
+
+S3 buckets must have versioning enabled, which ensures that a session log cannot
+be permanently altered or deleted. Teleport will always look at the oldest
+version of a recording.
+
+Below is an example of how to configure a Teleport auth server to store the
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:204]:
```suggestion
Below is an example of how to configure the Teleport Auth Service to store the
```
I've been capitalizing Teleport-specific product names so it's easier to separate them from general technical terms.

**Review by ptgott — APPROVED** [review]
Approved with a minor suggestion. Disabled auto-merge so you can apply it.

**Pull request merged**